### PR TITLE
[Merged by Bors] - chore: ensure all instances referred to directly have explicit names

### PR DESCRIPTION
**Pull request by kim-em**
Per https://github.com/leanprover/lean4/issues/2343, we are going to need to change the automatic generation of instance names, as they become too long.

This PR ensures that everywhere in Mathlib that refers to an instance by name, that name is given explicitly, rather than being automatically generated.

There are four exceptions, which are now commented, with links to https://github.com/leanprover/lean4/issues/2343.

This was implemented by running Mathlib against a modified Lean that appended `_ᾰ` to all automatically generated names, and fixing everything.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Prod.lean [hunk 1]
```diff
@@ -37,8 +37,8 @@ variable {A : Type _} {B : Type _} {G : Type _} {H : Type _} {M : Type _} {N : T
 
 namespace Prod
 
-@[to_additive]
-instance [Mul M] [Mul N] : Mul (M × N) :=
+@[to_additive instAddProd]
+instance instMulProd [Mul M] [Mul N] : Mul (M × N) :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/Group/Prod.lean:41]: Ideally these should be called `Prod.instAdd`  not `Prod.instAddProd`. Was this a deliberate choice, or just an artifact of some automation?

Comment (semorrison) [reply Mathlib/Algebra/Group/Prod.lean:41]: We're inconsistent this (i.e. does being in the namespace mean you leave off that name as a terminal fragment of the an instance name?) across Mathlib. This PR is not attempting to resolve the question of "how does Mathlib want to name things" or to match names to a not-yet-existing consensus about that question. It just ensures we never rely on a Lean 4 auto-generated name, to ensure that it is possible to make changes in Lean 4 without having to fight Mathlib.

Nevertheless, I'll change this file now.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
bors r-

bors d+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**